### ISSUE TITLE
Push notification doesn’t mark as read

### ISSUE THREAD

**Issue by Huddie** [GitHawkApp/GitHawk#2298]
Anyone else?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by j-f1** [comment]
I can repro this too.

**Comment by rnystrom** [comment]
Can you explain the issue a little more? What is supposed to be read?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by j-f1** [comment]
When you tap a push notification, the issue/PR isn’t marked as read in GitHawk’s inbox view.

**Comment by Huddie** [comment]
Ya what @j-f1 said. Push notifications open you to an issue but even though that means you’ve read the issue, the issue still shows in the inbox 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
Yeah it seems to perform the route but the reading logic is [here](https://github.com/GitHawkApp/GitHawk/blob/508525d892f8817914027e87c965eeb3936c6970/Classes/Notifications/NotificationSectionController.swift#L82) which is probably not being hit via the route.

**Comment by rnystrom** [comment]
Oh now I understand. If you have this enabled in settings, it'll work. But otherwise, it doesn't mark read unless you do it from the inbox.

Would you expect this to work as if you marked it read from inbox?

**Comment by Huddie** [comment]
I believe even with read on in settings it doesn’t read when opening via push. Anyone else able to confirm?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
I just confirmed that opening from push does not mark the notification as read. 

@rnystrom ya I feel like it should mark it as if read from inbox. Any reason why reading it via push is different from reading via inbox?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by iDjay** [comment]
Other question, you only get notifications when the app runs in the background, right? Otherwise I‘m having a bug. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
Yes thats correct 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
